apps/api/src/app/events/usecases/add-job/add-delay-job.usecase.ts

```diff
@@ -22,71 +21,6 @@ export class AddDelayJob {
 
     await this.jobRepository.updateStatus(command.organizationId, data._id, JobStatusEnum.DELAYED);
 
-    return await this.calculateDelayAmount(data);
-  }
-
-  private checkValidDelayOverride(data: JobEntity): boolean {
-    if (!data.overrides?.delay) {
-      return false;
-    }
-    const values = Object.values(DigestUnitEnum);
-
-    return (
-      typeof data.overrides.delay.amount === 'number' &&
-      values.includes(data.overrides.delay.unit as unknown as DigestUnitEnum)
-    );
-  }
-
-  private async calculateDelayAmount(data: JobEntity): Promise<number> {
-    if (!data.step.metadata) throw new ApiException(`Step metadata not found`);
-
-    if (data.step.metadata.type === DelayTypeEnum.SCHEDULED) {
-      const delayPath = data.step.metadata.delayPath;
-      if (!delayPath) throw new ApiException(`Delay path not found`);
-
-      const delayDate = data.payload[delayPath];
-
-      const delay = differenceInMilliseconds(new Date(delayDate), new Date());
-
-      if (delay < 0) {
-        throw new ApiException(`Delay date at path ${delayPath} must be a future date`);
-      }
-
-      const noIdenticalDelay = await this.noExistingDelayedJobForDate(data, delayPath, delayDate);
-
-      if (noIdenticalDelay) {
-        return delay;
-      }
-
-      return delay + 1000;
-    }
-
-    if (this.checkValidDelayOverride(data)) {
-      return AddJob.toMilliseconds(data.overrides.delay.amount as number, data.overrides.delay.unit as DigestUnitEnum);
-    }
-
-    return AddJob.toMilliseconds(data.step.metadata.amount as number, data.step.metadata.unit as DigestUnitEnum);
-  }
-
-  /**
-   * To handle case of Scheduled Delay (triggered multiple times with the exact same future date) followed by a Digest.
-   * To avoid duplicate pending digests which would result in duplicate messages sent.
-   */
-  private async noExistingDelayedJobForDate(
-    data: JobEntity,
-    currentDelayPath: string,
-    currentDelayDate: string
```

Comment:
Was not needed anymore. It is not happening, thanks to Pablo

Reply:
I think we should remove it only if we are 100% sure this issue won't happen and we have a test that can simulate the possible error otherwise I think it could be risky.

Reply:
The tests created for that use case passed, and I did multiple manual checks to make sure. 
I agree, I'm not crazy about removing it, but it really seems to be resolved